pkg/scanners/scannerv4/convert.go

```diff
@@ -191,50 +198,123 @@ func setScoresAndScoreVersion(vuln *storage.EmbeddedVulnerability, vulnCVSS *v4.
 		return nil
 	}
 	errList := errorhelpers.NewErrorList("failed to parse vector")
-	if v2 := vulnCVSS.GetV2(); v2 != nil {
-		if c, err := cvssv2.ParseCVSSV2(v2.GetVector()); err == nil {
-			err = cvssv2.CalculateScores(c)
-			if err != nil {
-				errList.AddError(fmt.Errorf("calculating CVSS v2 scores: %w", err))
-			}
-			// Use the report's score if it exists.
-			if baseScore := v2.GetBaseScore(); baseScore != 0.0 && baseScore != c.Score {
-				log.Debugf("Calculated CVSSv2 score does not match given base score (%f != %f) for %s. Using given score...", c.Score, baseScore, vuln.GetCve())
-				c.Score = baseScore
-			}
-			c.Severity = cvssv2.Severity(c.Score)
-			vuln.CvssV2 = c
-			// This sets the top level score for use in policies.
+	if vulnCVSS.GetV2() != nil {
+		score, sv, cvssV2, err := toCVSSV2Scores(vulnCVSS, vuln.GetCve())
+		if err != nil {
+			errList.AddError(err)
+		} else {
+			// This sets the top-level score for use in policies.
 			// It will be overwritten if v3 exists.
-			vuln.ScoreVersion = storage.EmbeddedVulnerability_V2
-			vuln.Cvss = c.Score
+			vuln.ScoreVersion = sv
+			vuln.Cvss = score
+			vuln.CvssV2 = cvssV2
+		}
+	}
+
+	if vulnCVSS.GetV3() != nil {
+		score, sv, cvssV3, err := toCVSSV3Scores(vulnCVSS, vuln.GetCve())
+		if err != nil {
+			errList.AddError(err)
 		} else {
-			errList.AddError(fmt.Errorf("v2: %w", err))
+			vuln.CvssV3 = cvssV3
+			// Overwrite CVSSV2 if already set.
+			vuln.ScoreVersion = sv
+			vuln.Cvss = score
 		}
 	}
-	if v3 := vulnCVSS.GetV3(); v3 != nil {
-		if c, err := cvssv3.ParseCVSSV3(v3.GetVector()); err == nil {
-			err = cvssv3.CalculateScores(c)
-			if err != nil {
-				errList.AddError(fmt.Errorf("calculating CVSS v3 scores: %w", err))
+
+	return errList.ToError()
+}
+
+func setScoresAndScoreVersionList(vuln *storage.EmbeddedVulnerability, cvssMetrics []*v4.VulnerabilityReport_Vulnerability_CVSS) error {
+	errList := errorhelpers.NewErrorList("failed to get CVSS Metrics")
+
+	if len(cvssMetrics) == 0 {
+		return nil
+	}
+
+	var scores []*storage.CVSSScore
+	for _, cvss := range cvssMetrics {
+		score := &storage.CVSSScore{
+			Source: storage.Source(cvss.Source.Number()),
+			Url:    cvss.Url,
+		}
+
+		if cvss.GetV3() != nil {
```

Comment:
mentioned before, but this is very similar to what we do in `setScoresAndScoreVersion`. If we plan on removing that one day, then this is ok, but otherwise, perhaps we should consider reusing some pieces of that function